packages/server/dataloader/azureDevOpsLoaders.ts

```diff
@@ -0,0 +1,425 @@
+  return new DataLoader<TeamUserKey, WorkItem[] | undefined, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const returnWorkItems = [] as WorkItem[]
+          const auth = await parent
+            .get('teamMemberIntegrationAuths')
+            .load({service: 'azureDevOps', teamId, userId})
+
+          if (!auth) {
+            return undefined
+          }
+
+          const provider = await parent.get('integrationProviders').loadNonNull(auth.providerId)
+
+          const manager = new AzureDevOpsServerManager(
+            auth,
+            provider as IntegrationProviderAzureDevOps
+          )
+
+          const restResult = await manager.getAllUserWorkItems(null, false)
+          const {error, workItems} = restResult
+          if (error !== undefined || workItems === undefined) {
+            console.log(error)
+          } else {
+            returnWorkItems.push(...workItems)
+          }
+          return returnWorkItems
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : undefined))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.teamId}:${key.userId}`
+    }
+  )
+}
+
+export const azureDevUserInfo = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, AzureUserInfo | undefined, string> => {
+  return new DataLoader<TeamUserKey, AzureUserInfo | undefined, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const auth = await parent
+            .get('teamMemberIntegrationAuths')
+            .load({service: 'azureDevOps', teamId, userId})
+
+          if (!auth) {
+            return undefined
+          }
+
+          const provider = await parent.get('integrationProviders').loadNonNull(auth.providerId)
+
+          const manager = new AzureDevOpsServerManager(
+            auth,
+            provider as IntegrationProviderAzureDevOps
+          )
+
+          const restResult = await manager.getMe()
+          const {error, azureDevOpsUser} = restResult
+          if (error !== undefined || azureDevOpsUser === undefined) {
+            console.log(error)
+            return undefined
+          }
+
+          return {
+            ...azureDevOpsUser
+          }
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : undefined))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.teamId}:${key.userId}`
+    }
+  )
+}
+
+export const allAzureDevOpsAccessibleOrgs = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, Resource[], string> => {
+  return new DataLoader<TeamUserKey, Resource[], string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const auth = await parent
```

Comment:
-1 Why not use `const auth = await parent.get('freshAzureDevOpsAuth').load({teamId, userId})`?